### DISCUSSION TITLE
Can I add to the UI?

### DISCUSSION THREAD

**Discussion by adegans** [endless-sky/endless-sky#7834] [General]
For my plugin ( https://github.com/adegans/SpaceCrate )  I need some more UI stuff and I think it should (or might) be possible since there is a Interfaces.txt setup. But I'm having trouble finding what I need.

**2 Main things I want to do:**

**1** 
I want to add a tab in the "Player Info" (Where you can see the ship list and such). At the top of that are 2 tabs. I want to add a 3rd. In that tab I want to show text/values/stats.

Possible? If so, how? Is there any documentation for it?

If not, can I add a tab/section to the Logbook? Like there is Factions and years now. Can I add my tab there?

**2** 
On entering a system there is always a line of text that says "for X or Y mission you need to land on Z planet."
I think it's triggered by an 'on enter' event.
Can I change the color of that so it's more visible? What override do I use to do that? Probably one of the interfaces.txt color values? 

Please advise... Thanks!

**Comment by quyykk** [comment]
The interfaces.txt is used to described the UI, so you can modify it but adding new buttons and such have to include modifications to the game itself unfortunately.

> I want to add a tab in the "Player Info" (Where you can see the ship list and such). At the top of that are 2 tabs. I want to add a 3rd. In that tab I want to show text/values/stats.

not possible, as mentioned

> If not, can I add a tab/section to the Logbook? Like there is Factions and years now. Can I add my tab there?

That is possible yes. See the creating missions page on the wiki (search for `log`).

> Can I change the color of that so it's more visible? What override do I use to do that? Probably one of the interfaces.txt color values?

That's technically possible, but you'd change the color of every line as well. Search for `message importance` in interfaces.txt to see the current values. Although, I think bumping the message importance of the "you need to land X" message makes sense.

**Reply by adegans** [reply]
Hmm none of this would work for my ideas. BUmmer.
The color change, perhaps such messages should be highlighted green (or something bright, blue?)  in future versions - Often they are lost in all the "scanning your ship" nonsense when entering a system.
Unlike the "Your fleet has attracted Pirates" on entering a system. But to show a positive as orange is not a good idea I think.
